src/utils/appInstaller/index.ts

```diff
@@ -0,0 +1,194 @@
+import { BuidlerRuntimeEnvironment } from '@nomiclabs/buidler/types'
+import { ethers } from 'ethers'
+import { APMRegistryInstance, KernelInstance } from '~/typechain'
+import { getLog } from '~/src/utils/getLog'
+import {
+  AppInstallerOptions,
+  AppInstaller,
+  AppOptions,
+  AppInstalled,
+  NetworkType
+} from '~/src/types'
+import {
+  toVersionArray,
+  getFullName,
+  namehash,
+  getContentHash,
+  utf8ToHex
+} from './utils'
+import getAbiFromContentUri from './getAbiFromContentUri'
+import getExternalRepoVersion from './getRepoVersion'
+import assertEnsDomain from './assertEnsDomain'
+
+const ANY_ENTITY = '0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF'
+
+/**
+ * Get an initialized instance of of appInstaller
+ * @param installerOptions
+ * @param bre
+ */
+/* eslint-disable-next-line @typescript-eslint/explicit-function-return-type */
+export default function AppInstaller(
+  installerOptions: AppInstallerOptions,
+  bre: BuidlerRuntimeEnvironment
+): AppInstaller {
+  /**
+   * @param name "finance"
+   */
+  return function appInstaller(
+    name: string,
+    appOptions?: AppOptions
+  ): Promise<AppInstalled> {
+    return _installExternalApp(
+      { name, ...(appOptions || {}), ...installerOptions },
+      bre
+    )
+  }
+}
+
+/* eslint-disable-next-line @typescript-eslint/explicit-function-return-type */
+async function _installExternalApp(
+  {
+    name,
+    version,
+    network = 'homestead',
+    initializeArgs,
+    skipInitialize,
+    apmAddress,
+    dao,
+    ipfsGateway
+  }: {
+    name: string
+    version?: string
+    initializeArgs?: any[]
+    skipInitialize?: boolean
+    network?: NetworkType
+    apmAddress: string
+    dao: KernelInstance
+    ipfsGateway: string
+  },
+  bre: BuidlerRuntimeEnvironment
+) {
+  const provider = new ethers.providers.InfuraProvider(network)
```

Comment:
We should probably used the declared mainnet network in the `arapp.json` file if available

Reply:
We will address this in a new PR